### PR TITLE
fix(Input): replace disabled class with disabled attribute

### PR DESCRIPTION
**Pull request by dvdzkwsk**
Resolves https://github.com/Semantic-Org/Semantic-UI-React/issues/1565.

@levithomason This, ironically enough, is a matter of semantics. Your call here, but in my opinion this improves the functionality of inputs as they now receive an actual `disabled` attribute instead of relying on custom styling with a `disabled` class above the input element itself. This allows disabled inputs to inherit standard browser behavior and conform to general user expectations.

I checked it out in the docs and by removing the `disabled` class, the styles still look correct.

### CODE REVIEW

File: src/lib/htmlInputPropsUtils.js [hunk 1]
```diff
@@ -5,12 +5,8 @@ export const htmlInputAttrs = [
   'selected', 'defaultValue', 'defaultChecked',
 
   // LIMITED HTML PROPS
-  'autoCapitalize', 'autoComplete', 'autoCorrect', 'autoFocus', 'checked', 'form', 'max', 'maxLength', 'min',
+  'autoCapitalize', 'autoComplete', 'autoCorrect', 'autoFocus', 'checked', 'disabled', 'form', 'max', 'maxLength', 'min',
   'multiple', 'name', 'pattern', 'placeholder', 'readOnly', 'required', 'step', 'type', 'value',
-
-  // Heads Up!
```
Comment (dvdzkwsk) [inline src/lib/htmlInputPropsUtils.js]: Handled this by not applying the `disabled` class. Potentially breaking change for users who relied on that as a CSS selector.

**Comment by levithomason** [comment]
Per offline chats, let's keep both the top level className so child elements are disabled but also apply the attribute.

I'll PR the CSS to handle the double opacity issue.

**Comment by levithomason** [comment]
Tracking https://github.com/Semantic-Org/Semantic-UI/pull/5284

**Comment by levithomason** [comment]
This seems to have broken tests for `input shorthand prop (common)`.

**Comment by dvdzkwsk** [comment]
SUI issue has no traction :(

File: src/elements/Input/Input.js [hunk 2]
```diff
@@ -135,6 +142,23 @@ class Input extends Component {
...
+    const tabIndex = this.computeTabIndex()
+    const unhandled = getUnhandledProps(Input, this.props)
+    const [htmlInputProps, rest] = partitionHTMLInputProps(unhandled)
+
+    htmlInputProps.ref = this.handleInputRef
+    htmlInputProps.type = type
+
+    if (disabled) htmlInputProps.disabled = disabled
+    if (onChange) htmlInputProps.onChange = this.handleChange
+    if (tabIndex) htmlInputProps.tabIndex = tabIndex
```
Comment (layershifter) [inline src/elements/Input/Input.js:157]: The problem is that the component receives `disabled` as `undefined`, same is for `tabIndex`

I made tests working, however, it's not the best idea :hankey:  May be better solution possible there?

Comment (levithomason) [reply src/elements/Input/Input.js:157]: I believe if the `disabled` prop is `undefined`, then the `htmlInputProps.disabled` should also be undefined.  Perhaps  I'm overlooking the issue or use case here.

Comment (layershifter) [reply src/elements/Input/Input.js:157]: @levithomason You're right, but if we replace current code with:

```js
if (onChange) htmlInputProps.onChange = this.handleChange

return [{...htmlInputProps, disabled, tabIndex}, rest]
```

But, we will have failing tests of shorhand.

Comment (layershifter) [reply src/elements/Input/Input.js:157]: However, we can fix this in test:
```diff
  common.implementsHTMLInputProp(Input, {
    alwaysPresent: true,
-    shorthandDefaultProps: { type: 'text' },
+    shorthandDefaultProps: { disabled: undefined, tabIndex: undefined, type: 'text' },
  })
```

May be I overlooked something?

**Comment by levithomason** [comment]
Thanks for the updates @layershifter !

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.68.3`.

**Pull request merged**